masq_lib/src/test_utils/mock_websockets_server.rs

```diff
@@ -490,120 +532,205 @@ mod tests {
-        let second_broadcast_trigger = UiBroadcastTrigger {
-            number_of_broadcasts_in_one_batch: None,
-            position_to_send_the_signal_opt: None,
-        };
-        connection.send(second_broadcast_trigger.clone());
+        //sending another broadcast trigger (unlimited) to get the third, fourth and sixth message
+        connection.send(broadcast_trigger_two_with_no_limit);
+        //finally, when using the trigger again, we can get three other messages
 
-        //TESTED BY COMPLETING THE TASK - NO ADDITIONAL ASSERTION NEEDED
-        let _first_broadcast_from_the_second_batch: UiNewPasswordBroadcast =
-            connection.receive().unwrap();
-        //testing that the second arriving is really the last one; that means that used broadcasts are removed
-        let _second_broadcast_from_the_second_batch: UiNodeCrashedBroadcast =
-            connection.receive().unwrap();
+        let _ = (0..3)
+            .map(|_| connection.receive().unwrap())
+            .collect::<Vec<UiNewPasswordBroadcast>>();
 
-        let requests = stop_handle.stop();
-        assert_eq!(
-            *requests[0].as_ref().unwrap(),
-            first_broadcast_trigger.tmb(0)
+        //here we shouldn't be able to jump over to some other broadcast in the queue though there is one!;
+        //instead we should see an error because next we meet a conversational message
+        let naive_attempt_number_three_to_receive_another_broadcast_from_the_queue: Result<
+            UiNodeCrashedBroadcast,
+            (u64, String),
+        > = connection.receive();
+
+        let _received_message_number_seven: UiDescriptorResponse = connection
+            .transact_with_context_id(conversation_number_three_request, 3)
+            .unwrap();
+        //we want to get to the last broadcast
+        connection.send(broadcast_trigger_three_with_no_limit);
+
+        let _received_message_number_eight: UiNodeCrashedBroadcast = connection.receive().unwrap();
+        //the queue should be empty now
+
+        let naive_attempt_number_four: Result<UiNodeCrashedBroadcast, (u64, String)> =
+            connection.receive();
+        //the previous attempt eliminated the possibility of another broadcast
+        //but what happens when new conversation tried
+
+        let naive_attempt_number_five: Result<UiDescriptorResponse, (u64, String)> =
+            connection.transact_with_context_id(UiDescriptorRequest {}, 0);
+
+        let _ = stop_handle.stop();
+        ////////////////////////////////////////////////////////////////////////////////////////////
+        //assertions for deliberately caused errors
+        let error_message_number_one = naive_attempt_number_one_to_receive_the_third_broadcast
+            .unwrap_err()
+            .1;
+        assert!(
+            error_message_number_one.contains(
+                "Expected a corresponding response pulled out from the queue. \
```

Comment:
Make it clear that it was a broadcast (not a conversation) that we were looking for and couldn't find.

Reply:
This is a problem. This message actually doesn't arrive from the server - it's generated on the client's end - and technically means that we've waited too long and aborted from that process otherwise it'd have blocked forever.